### ISSUE TITLE
Ratelimiting

### ISSUE THREAD

**Issue by Siebe** [Podcastindex-org/docs-api#30]
So, like I told @daveajones , I found out there's a rate limiting system in place... the hard way :}
So thank so much for the warning and plz update docs <3

And also:

As a developer, when I create an service on the API, I would like to be informed on the current status of my requests and limits, so I can avoid hitting that 429 or predict when I am able to send new requests.

Normally this is done through headers, plus some explanation in the documentation on how to interpret these headers. A countdown header and knowing cooldown/reset time would suffice I think. 
Protip: custom headers usually start with x- and I have a feeling that might help pushing them through Cloudflare.

**Comment by picard102** [comment]
What is the rate limit set to and for what period?

**Comment by cbarlow26** [comment]
@daveajones said, "You want to keep your requests at less than one per second" (#16 ). Is this per account, per key, or per client?

**Comment by daveajones** [comment]
We don't publish rate limits because we have no SLA's or guarantees.  We are a free project that runs on donations and I don't want everyone pushing their usage right up to the edge of the published rate limits.  If everyone does that we can't support the load.  We adjust things continuously as needed to support the load we have at the time.

**Comment by daveajones** [comment]
No. A rate limit is a 429. Make sure you are using a proper user agent header and not the ones from the sample code.

**Comment by SteveCutler** [comment]
> No. A rate limit is a 429. Make sure you are using a proper user agent header and not the ones from the sample code.

I think I'm doing everything right, I had it working and then all of a sudden it stopped.
This is my request code, I don't know if anything jumps out to you when you look at this:

const headerTime = Math.floor(Date.now() / 1000).toString()
            const key = import.meta.env.VITE_PODCASTINDEX_KEY
            const secret = import.meta.env.VITE_PODCASTINDEX_SECRET
            const hashInput = key + secret + headerTime

            const hash = CryptoJS.SHA1(hashInput).toString(CryptoJS.enc.Hex)
           
            const episodesResponse = await axios.get(
                `https://api.podcastindex.org/api/1.0/episodes/byfeedid?id=${id}`,
                {
                    headers: {
                        'User-Agent': 'AppleMusicDashboard/1.0',
                        'X-Auth-Key': key,
                        'X-Auth-Date': headerTime,
                        Authorization: hash,
                    },
                    params: {
                        fulltext: true,
                        max: 10,
                    },
                }
            )
            console.log('episodes', episodesResponse)
            

all the variables print out fine when I log them to console, but I'm getting the hash auth error:
The hash in the Authorization header doesn't match up.  Please check your hash compute code: https://podcastindex-org.github.io/docs-api/#overview--authentication-details

**Comment by SteveCutler** [comment]
Resolved by making a new key secret pair. Maybe I didn't copy/paste them correctly last time..

**Comment by jtgi** [comment]
> Resolved by making a new key secret pair. Maybe I didn't copy/paste them correctly last time..

One gotcha with the api key is you need to escape `$`, it has special meaning in most environments.

```diff
# .env
- PODCAST_INDEX_API_SECRET="ejEz9kjjX^YaSRbBT3GlowfMRR6C6QFW4nxGNc$t"
+ PODCAST_INDEX_API_SECRET="ejEz9kjjX^YaSRbBT3GlowfMRR6C6QFW4nxGNc\$t"
```

**Comment by ueen** [comment]
What would be the correct way to serve users with a podcatcher app? cache the podcastindex on my server somehow or use the api? is there a special one for developers that serves users or something?

**Comment by daveajones** [comment]
Depends on what platform the app runs on and whether you will have back-end servers supporting the app or if it's on device.  More info on the stack is helpful to give an answer.

**Comment by ueen** [comment]
It'll be a podcatcher for Android and iOS, released on the app stores, ideally I wouldn't run a server at all, but I also don't want to rely on the iTunes or other proprietary APIs.

**Comment by daveajones** [comment]
Keeping an on-device follow/subscribe database will be your best bet and then just use Podcast Index/Apple Podcasts for search/subscribe.  Castamatic/Antennapod are good examples of that method.
